engine/src/main/java/com/redhat/cloud/notifications/exports/ExportsEventListener.java

```diff
@@ -0,0 +1,335 @@
+                    this.exportsService.notifyErrorExport(this.exportServicePsk, exportRequestUuid, application, resourceUuid, exportError);
+                }
+            });
+        }
+
+        this.meterRegistry.counter(EXPORTS_SERVICE_SUCCESSES_COUNTER).increment();
+
+        return message.ack();
+    }
+
+    /**
+     * Checks if the provided resource type is handleable.
+     * @param resourceType the resource type to be checked.
+     * @return true if the provided resource type is handleable.
+     */
+    boolean isValidResourceType(final String resourceType) {
+        return RESOURCE_TYPE_EVENTS.equals(resourceType);
+    }
+
+    /**
+     * Checks if the provided Cloud Event comes from the "export-service" and
+     * if the event type is a proper export request.
+     * @param cloudEvent the cloud event to check.
+     * @return true if the cloud event comes from the exports service, and it is
+     * of the "export request" type.
+     */
+    boolean isAnExportRequest(final ConsoleCloudEvent cloudEvent) {
+        return EXPORT_SERVICE_URN.equals(cloudEvent.getSource())
+            && CE_EXPORT_REQUEST_TYPE.equals(cloudEvent.getType());
+    }
+
+    /**
+     * Extracts the date from the provided filter. It also looks if the given
+     * date is valid, in the sense of it not being older than a month or in the
+     * future. {@code null} dates are considered valid, since that means that
+     * there is no filter to apply to the queries.
+     * @param filters the map of filters.
+     * @param filterName the name of the filter of the date to extract.
+     * @return the parsed date or {@code null} if there is no date in the
+     * specified filter.
+     */
+    LocalDate extractDateFromFilter(final Map<String, Object> filters, final String filterName) {
+        final Object dateAsString = filters.get(filterName);
+
+        if (dateAsString == null) {
+            return null;
+        }
+
+        final LocalDate parsedDate = LocalDate.parse((String) dateAsString);
+
+        final LocalDate today = LocalDate.now(ZoneOffset.UTC);
+        if (parsedDate.isAfter(today)) {
+            throw new IllegalStateException("the specified date is in the future");
+        }
+
+        final LocalDate aMonthAgo = today.minusMonths(1);
+        if (aMonthAgo.isAfter(parsedDate)) {
+            throw new IllegalStateException("the specified date is older than a month");
+        }
+
+        return parsedDate;
+    }
+
+    /**
+     * Extracts the export request's {@link UUID}, which comes in the Cloud
+     * Event's subject. Beware that the Cloud Event's {@link UUID} is not the
+     * same as the export request's {@link UUID}, or the resource's
+     * {@link UUID}.
+     * @param subject the received subject of the Cloud Event.
+     * @return the extracted {@link UUID} from the subject.
+     */
```

Comment:
Is there any reason why the request UUID is not used as the cloud event UUID?

Why is a resource UUID needed in addition to the request UUID?

Reply:
There are three UUIDs:

- The Cloud Event message's UUID, which represents that unique message itself.
- The subject's UUID, which represents an export request.
- The resource's UUID, which is a specific data request from that export request.

The export request could be seen as the parent of the resource: one export request may require multiple resource requests from multiple applications. So, maybe that export request's UUID can come in the payload too to make it easier to parse? We can ask for that to the export team, I think.

Reply:
It looks like the Cloud Event UUID and the subject (or request) UUID are two identifiers for the same thing. If so, I'd use the same UUID for both. But that's not a decision to make on our side anyway 😃

Thanks for clarifying the resource UUID, it makes sense to me now.